### PR TITLE
BUG: Error when filling a value with parentheses #2268

### PR DESCRIPTION
**Pull request by KanorUbu**
Code to fix issue #2268

### CODE REVIEW

File: pypdf/_writer.py [hunk 1]
```diff
@@ -920,7 +920,8 @@ def _update_text_field(self, field: DictionaryObject) -> None:
         else:  # /Tx
             txt = field.get("/V", "")
             sel = []
-
+        # Escape parentheses (Ref. 7.3.4.2 Literal Strings)
```
Comment (pubpub-zz) [inline pypdf/_writer.py:923]:
```suggestion
        # Escape parentheses (pdf 1.7 reference, table 3.2  Literal Strings)
```

File: pypdf/_writer.py [hunk 2]
```diff
@@ -920,7 +920,8 @@ def _update_text_field(self, field: DictionaryObject) -> None:
...
+        txt = txt.replace('(', r'\(').replace(')', r'\)')
```
Comment (pubpub-zz) [inline pypdf/_writer.py:924]: \ should also be covered
```suggestion
        txt = txt.replace("\\", "\\\\").replace("(", r"\(").replace(")", r"\)")
```

**Review by pubpub-zz — CHANGES_REQUESTED** [review]
thanks for your contribution.
2 comments to complete your work

**Comment by KanorUbu** [comment]
I've considered your feedback

**Review by pubpub-zz — APPROVED** [review]

**Comment by stefan6419846** [comment]
Shouldn't we add a test for this as well, at least at a basic level?

**Comment by pubpub-zz** [comment]
> Shouldn't we add a test for this as well, at least at a basic level?

I've thought about it but the only way is a visual check 🫤. If the test consist in checking replace is working I don't think it worth it

**Comment by MartinThoma** [comment]
Thank you for your contribution @KanorUbu :pray: 

If you want, I'll add you to https://pypdf.readthedocs.io/en/latest/meta/CONTRIBUTORS.html

**Comment by MartinThoma** [comment]
And thanks @pubpub-zz / @stefan6419846 for the review :-)

**Pull request merged**